### ISSUE TITLE
Schema-annotation: add property  `addJavaDoc`

### ISSUE THREAD

**Issue by nimo23** [smallrye/smallrye-open-api#2090]
Currently, I need to duplicate existing Javadoc from the property into the `Schema.description` tag. This is very annoying and hurts DRY (don’t repeat yourself). It fills the Java class with redundant data and is worse for maintenance, etc..:

	/**
	 * The description.
	 * 
	 * <p>
	 * Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
	 * eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
	 * voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
	 */
	@Schema(readOnly = true, addJavaDoc=true, description = "The description. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy\n"
			+ "	 * eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam\n"
			+ "	 * voluptua. At vero eos et accusam et justo duo dolores et ea rebum.")
	private final String description;

Please add a new property for the `Schema`-Tag, e.g. `addJavaDoc = true | false`. If the value is set to true, the Javadoc is appended to the OpenAPI `description` value:

For example:

	/**
	 * The description.
	 * 
	 * <p>
	 * Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
	 * eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
	 * voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
	 */
	@Schema(readOnly = true, addJavaDoc=true)
	private final String description;

another example using both `addJavaDoc` and an explicit `description` that simply appends the Javadoc after the description value (in this case: "My description. The description page..."):

	/**
	 * The description.
	 * 
	 * <p>
	 * Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
	 * eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
	 * voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
	 */
	@Schema(readOnly = true, addJavaDoc=true, description="My description")
	private final String description;

The Javadoc can contain html-tags or Markdown-Syntax (https://openjdk.org/jeps/467) which needs to be considered when putting this in the openapi `description` value.

**Comment by cristalp** [comment]
In my company, we check with CheckStyle that methods have full Javadoc.

However, when a method is well documented with OpenAPI, that's not necessary (DRY, as you stated).

So we just annotate the following:
```java
@SuppressWarnings("javadoc") // Documented with OpenAPI
```

**Comment by MikeEdgar** [comment]
I'm pretty sure this issue has been raised before. The main problem is that when scanning annotations, the source code (and JavaDoc comments) are no longer available. There might be a way to do this in the Maven plugin where the comments are preprocessed, but I haven't seen any obvious solution yet.

**Comment by cristalp** [comment]
We don't document private properties, we document getters and setters, since they are public.

**Comment by MikeEdgar** [comment]
> > There might be a way to do this in the Maven plugin where the comments are preprocessed, but I haven't seen any obvious solution yet.
> 
> This would be a valuable solution. Imagine a `Smallrye annotation preprocessor` on which compilation (the generation of `openapi.yml`) is done where it has access to the source code (**similar** to the preprocessing of `JBoss Message Logger`, `Hibernate Criteria Builder` or `Jakarta Data`, which also creates some files (in this case Java files) live within `target/generated/sources`-package).
> 
> I think when generating from Java to OpenAPI such important things (like setting the `description` via `JavaDoc`) should be taken into account.

Yes, it seems feasible [1] using an annotation processor, or the Maven `process-sources` or `generate-resources` phase.

[1] https://docs.oracle.com/en/java/javase/17/docs/api/java.compiler/javax/lang/model/util/Elements.html#getDocComment(javax.lang.model.element.Element)

**Comment by MikeEdgar** [comment]
@nimo23 I won't have the time for it anytime soon due to competing priorities. We always welcome contributions from the community if you have the spare cycles to give it a try.
